modules/ROOT/pages/6.0-release-notes-events.adoc

```diff
@@ -0,0 +1,100 @@
+= TinyMCE 6.0 Event changes
+:navtitle: TinyMCE 6.0 Event changes
+:description: TinyMCE 6.0 Event changes
+:keywords: releasenotes, events
+
+This section documents the improvements and fixes made to {productname} 6.0 Events.
+
+* xref:events[Events]
+* xref:new-and-improved-events][New and improved Events]
+* xref:fixed-events[Fixed Events]
+
+// tag::events[]
+[[events]]
+== Events
+
+
+[[new-and-improved-events]]
+=== New and improved events
+
+
+[[autocompleter]]
+==== `AutocompleterStart`, `AutocompleterUpdate` and `AutocompleterEnd`
+
+As part of moving `autocompleter` from the default `silver` theme to the {productname} core, three new events are now available: `AutocompleterStart`, `AutocompleterUpdate` and `AutocompleterEnd`.
+
+Themes can act on the `AutocompleterStart` and `AutocompleterUpdate` events and can call `AutocompleterEnd` to cancel the auto completer.
+
+
+[[before-set-content]]
+==== `BeforeSetContent`
+
+The `BeforeSetContent` event now includes the actual serialized content when passing in an `AstNode` to the `editor.setContent`.
+
+
+[[get-content]]
+==== `GetContent`
+
+The `GetContent` event now always passes a `string` for the `content` property.
+
+
+[[theme-load-error]]
+==== `ThemeLoadError`
+
+A `ThemeLoadError` event is now fired if the theme fails to load.
+
+
+[[fixed-events]]
+=== Fixed events
+
+
+[[before-input]]
+==== `beforeinput`
+
+The `beforeinput` event is expected to fire whenever material is added to or removed from a {productname} instance.
+
+Previously, however, because of overrides in the `DeleteBackspaceKeys.ts` module, the `beforeinput` event did not always fire when expected.
+
+With this release, these over-rides no longer prevent `beforeinput` events firing when expected.
+
+
+[[change]]
+==== `change`
+
+Previously, the `change` event was triggered by the first modification made to a {productname} editor instance.
+
+However, subsequent changes did not cause the event to fire until focus was switched away from the editor.
+
+For example, typing a single character in to the {productname} editor text-entry area triggered the `change` event.
+
+Consequent typing, or selecting of the entered character and modifying it by setting it to *bold* or _italic_, did not trigger further `change` events.
+
+Making further modifications and then switching focus away from the editor did trigger this event, however.
+
+As of {productname} 6, the `change` event is not fired unless and until
+
+. focus is switched away from the editor; and
+. changes have been made in the editor since focus was switched to it.
+
+NOTE: it is still possible to listen for modification to a {productname} editor instance before focus is changed. To do this, listen for the `dirty` event, rather than the `change` event.
```

Comment:
```suggestion
NOTE: It is still possible to listen for modification to a {productname} editor instance before focus is changed. To do this, listen for the `dirty` event, rather than the `change` event.
```